pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -774,6 +774,20 @@ type ValueFrom struct {
 	Expression string `json:"expression,omitempty" protobuf:"bytes,8,rep,name=expression"`
 }
 
+func (p *Parameter) HasValue() bool {
+	return p.Value != nil || p.Default != nil || p.ValueFrom != nil
+}
+
+func (p *Parameter) GetValue() string {
+	if p.Value != nil {
+		return p.Value.String()
+	}
+	if p.Default != nil {
+		return p.Default.String()
+	}
+	return ""
```

Comment:
should this return ValueFrom?

Reply:
`ValueFrom` is mainly for `input parameters` and it will be resolved during the runtime in the `ProcessArgs` function

Reply:
this is not consistent with `HasValue`? ought to be?